cartography/data/jobs/cleanup/kubernetes_import_cleanup.json

```diff
@@ -0,0 +1,60 @@
+{
+    "statements": [
+        {
+            "query": "MATCH (c:KubernetesContainer) WHERE c.lastupdated <> {UPDATE_TAG} WITH c LIMIT {LIMIT_SIZE} DETACH DELETE (c) return COUNT(*) as TotalCompleted",
+            "iterative": true,
+            "iterationsize": 100
+        },
+        {
+            "query": "MATCH (p:KubernetesPod) WHERE p.lastupdated <> {UPDATE_TAG} WITH p LIMIT {LIMIT_SIZE} DETACH DELETE (p) return COUNT(*) as TotalCompleted",
+            "iterative": true,
+            "iterationsize": 100
+        },
+        {
+            "query": "MATCH (s:KubernetesNamespace) WHERE s.lastupdated <> {UPDATE_TAG} WITH s LIMIT {LIMIT_SIZE} DETACH DELETE (s) return COUNT(*) as TotalCompleted",
+            "iterative": true,
+            "iterationsize": 100
+        },
+        {
+            "query": "MATCH (c:KubernetesCluster) WHERE c.lastupdated <> {UPDATE_TAG} WITH c LIMIT {LIMIT_SIZE} DETACH DELETE (c) return COUNT(*) as TotalCompleted",
```

Comment:
As written, this will delete all clusters regardless of namespace.

You may want to consider cleaning up only the clusters that are part of the current namespace being processed.

Edit: ah I see that you run cleanup only once at the very end for all assets. In that case this is fine then.